### PR TITLE
Typescript error squashing

### PR DESCRIPTION
**Pull request by TomPretty**
## What are you doing in this PR?
I did some error squashing in:

- support-frontend/assets/helpers/campaigns/campaigns.ts
- support-frontend/assets/helpers/tracking/acquisitions.ts
- support-frontend/assets/pages/contributions-landing/components/ContributionFormContainer.tsx
- support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYou.tsx

We should probably do some more work to refactor `acquisitions.ts` as the types there are quite awkward - lots of `any`s & `A | null | undefined` which we would be better of to commiting to `A | null` or `A?`

### CODE REVIEW

File: support-frontend/assets/helpers/tracking/acquisitions.ts [hunk 1]
```diff
@@ -111,26 +125,17 @@ function storeReferrerAcquisitionDataInSessionStorage(
...
-		response.push({
-			name: participation,
-			variant: participations[participation as any],
-		});
-	});
-	return response;
+	return Object.entries(participations).map(([name, variant]) => ({
+		name,
+		variant,
+	}));
```
Comment (TomPretty) [inline support-frontend/assets/helpers/tracking/acquisitions.ts:138]: I'm not quite sure the history of these functions, but I think now they're best of being `map`s!

File: support-frontend/typescript-errors.json [hunk 2]
```diff
@@ -4881,6 +4784,12 @@
             "message": "A required parameter cannot follow an optional parameter."
         }
     ],
+    "TS2344": [
+        {
+            "path": "assets/helpers/page/page.ts",
+            "message": "Type 'A' does not satisfy the constraint 'Action<any>'."
+        }
+    ],
```
Comment (TomPretty) [inline support-frontend/typescript-errors.json:4792]: I'm not clear on how I introduced this one

File: support-frontend/assets/helpers/tracking/acquisitions.ts [hunk 3]
```diff
@@ -139,12 +144,15 @@ function buildReferrerAcquisitionData(
 ): ReferrerAcquisitionData {
 	// This was how referrer pageview id used to be passed.
 	const referrerPageviewId =
-		acquisitionData.referrerPageviewId || getQueryParameter('REFPVID');
+		(acquisitionData.referrerPageviewId as string | undefined) ??
+		getQueryParameter('REFPVID');
+
```
Comment (TomPretty) [inline support-frontend/assets/helpers/tracking/acquisitions.ts:149]: I think the real solution here is to get rid of the `Record<string, any>` type, but that feels like a real rabbit whole to fall down that I didn't want to tackle right now!

File: support-frontend/assets/helpers/tracking/acquisitions.ts (hunk 1 above)
Comment (tomrf1) [reply support-frontend/assets/helpers/tracking/acquisitions.ts:138]: 💯

**Review by tomrf1 — APPROVED** [review]
great work! That `acquisitionData` type needs some thought...

**Pull request merged**